public/js/modules/filter.js

```diff
@@ -1,67 +1,119 @@
-    this.clear.addEventListener('click', this.handleClear);
+    if (this.sortSearch) {
+      this.sortSearch.addEventListener("keyup", this.handleSearchChange);
+    }
+
+    if (this.sortCat) {
+      this.sortCat.addEventListener("change", this.handleChange);
+    }
+
+    if (this.sortType) {
+      this.sortType.addEventListener("change", this.handleChange);
+    }
+
+    this.clear.addEventListener("click", this.handleClear);
   }
 
   handleClear = () => {
-    this.select.value = "";
-    this.filterSort(this.select.value)
+    this.sortSearch ? this.sortSearch.value = "" : null;
+    this.sortCat ? this.sortCat.value = "" : null;
+    this.sortType ? this.sortType.value = "" : null;
+    this.filterSort("", "", "");
+  };
+
+  handleSearchChange = (e) => {
+    clearTimeout(this.timeout)
+
+    this.timeout  = setTimeout(() => {
+      this.handleChange()
+    }, 750)
   }
 
   handleChange = (e) => {
-    this.filterSort(e.target.value)
-  }
+    const searchValue = this.sortSearch ? this.sortSearch.value.toLowerCase() : "";
+    const catValue = this.sortCat ? this.sortCat.value : "";
+    const typeValue = this.sortType ? this.sortType.value : "";
 
-  filterSort(filterTerm) {
-    // build array of matches so we know how many we have
-    const matches = Array.prototype.filter.call(this.entries, function(entry) {
-      if (filterTerm === "all" || filter === "") {
-        return entry
-      }
+    this.filterSort(searchValue, catValue, typeValue)
+  };
 
-      return entry.dataset.filterTerms.includes(filterTerm);
-    })
+  filterSort(filterSearch, filterCat, filterType) {
+    // build array of matches
+    const matches = Array.prototype.filter.call(
+      this.entries,
+      function (entry) {
+        if (filterSearch === "" && filterCat === "" && filterType === "") {
+          return entry;
+        }
+
+        let searchMatch = true;
+        let catMatch = true;
+        let typeMatch = true;
+
+        if (filterSearch !== "") {
+          searchMatch = entry.querySelector("[data-filter-entry-text]").innerHTML.toLowerCase().includes(filterSearch)
+        }
+
+        if (filterCat !== "") {
+          catMatch = entry.dataset.filterCategories.includes(filterCat);
+        }
+
+        if (filterType !== "") {
+          typeMatch = entry.dataset.filterType.includes(filterType);
+        }
 
-    this.entries.forEach((entry) => {
-      // fade all out, deactivate all, then activate and fade in the matching entries
-      entry.animate(this.fadeOut, this.defaultTiming).finished.then(() => {
-        entry.dataset.active = false;
-        matches.forEach((match) => {
-          match.dataset.active = true
-          match.animate(this.fadeIn, this.defaultTiming)
-        })
+        return searchMatch && catMatch && typeMatch
+      }
+    );
+
+    this.sections.forEach((section) => {
+      section.animate(this.fadeOut, this.defaultTiming).finished.then(() => {
```

Comment:
Surprisingly, the initial slow down was from the animations, not the filtering, so this was changed to animate the entire section, rather than individual entries.